tests/app/dao/test_provider_details_dao.py

```diff
@@ -122,6 +125,71 @@ def test_get_alternative_sms_provider_fails_if_unrecognised():
         get_alternative_sms_provider('ses')
 
 
+@freeze_time('2016-01-01 00:30')
+def test_adjust_provider_priority_sets_priority(
+    restore_provider_details,
+    notify_user,
+    mmg_provider,
+):
+    # need to update these manually to avoid triggering the `onupdate` clause of the updated_at column
+    ProviderDetails.query.filter(ProviderDetails.identifier == 'mmg').update({'updated_at': datetime.min})
+
+    _adjust_provider_priority(mmg_provider, 50)
+
+    assert mmg_provider.updated_at == datetime.utcnow()
+    assert mmg_provider.created_by.id == notify_user.id
+    assert mmg_provider.priority == 50
+
+
+@freeze_time('2016-01-01 00:30')
+def test_adjust_provider_priority_adds_history(
+    restore_provider_details,
+    notify_user,
+    mmg_provider,
+):
+    # need to update these manually to avoid triggering the `onupdate` clause of the updated_at column
+    ProviderDetails.query.filter(ProviderDetails.identifier == 'mmg').update({'updated_at': datetime.min})
+
+    old_provider_history_rows = ProviderDetailsHistory.query.filter(
+        ProviderDetailsHistory.id == mmg_provider.id
+    ).order_by(
+        desc(ProviderDetailsHistory.version)
+    ).all()
+
+    _adjust_provider_priority(mmg_provider, 50)
+
+    updated_provider_history_rows = ProviderDetailsHistory.query.filter(
+        ProviderDetailsHistory.id == mmg_provider.id
+    ).order_by(
+        desc(ProviderDetailsHistory.version)
+    ).all()
+
+    assert len(updated_provider_history_rows) - len(old_provider_history_rows) == 1
+    assert updated_provider_history_rows[0].version - old_provider_history_rows[0].version == 1
+    assert updated_provider_history_rows[0].priority == 50
+
+
+@freeze_time('2016-01-01 00:30')
+def test_get_sms_providers_for_update_returns_providers(restore_provider_details):
+    six_minutes_ago = datetime(2016, 1, 1, 0, 24)
```

Comment:
Optional: The choice of five minutes for the treshold is fine but wondering if instead using one hour would make this test more reflective of the rest of the code and reduces a bit of the time needed to figure out what this test is testing. Could also change test name to something like `test_get_sms_providers_for_update_returns_providers_when_not_edited_within_threshold` or similar